File: BE/ytt/src/main/java/com/example/ytt/domain/auth/dto/SignUpDto.java
package com.example.ytt.domain.auth.dto;

import com.example.ytt.domain.user.dto.Role;
import com.fasterxml.jackson.annotation.JsonIgnore;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.Pattern;
import lombok.*;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class SignUpDto {

    @NotBlank
    private String email;

    @NotBlank
    @Pattern(regexp = "(?=.*[0-9])(?=.*[a-zA-Z])(?=.*\\W)(?=\\S+$).{12,20}", message = "비밀번호는 12~20자 영문 대 소문자, 숫자, 특수문자를 사용하세요.")
    private String password;

    @NotBlank
    private String name;

    @NotBlank
    @Pattern(regexp = "^\\d{2,3}-\\d{3,4}-\\d{4}$", message = "핸드폰 번호의 양식과 맞지 않습니다.")
    private String phoneNumber;

    @JsonIgnore
    private Role role = Role.CUSTOMER;
}

File: FE/app/src/main/java/com/wonchihyeon/ytt_android/data/model/SignUpDTO.kt
package com.wonchihyeon.ytt_android.data.model

data class SignUpDTO(
    val email: String,
    val password: String,
    val name: String,
    val phoneNumber: String,
    val role: Role,
    val accessToken: String? = null,   // 서버 응답에서 포함되는 Access Token
    val refreshToken: String? = null   // 서버 응답에서 포함되는 Refresh Token
)
